addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +113,144 @@ def _get_duration_from_tracking(self, trackings):
+            ]
+        return [fields.Domain.FALSE]
+
+    def _is_rotting_enabled(self):
+        return hasattr(self, '_track_duration_field') and 'rotting_threshold_days' in self[self._track_duration_field] and (
+            not self  # api.model call
+            or any(stage.rotting_threshold_days for stage in self[self._track_duration_field])
+        ) and 'date_last_stage_update' in self
+
+    @api.depends(lambda self: self._get_rotting_depends_fields())
+    def _compute_rotting(self):
+        could_rot_records = self.filtered_domain(self._get_rotting_domain())
+        clean_records = (self - could_rot_records)
+        for stage, records in could_rot_records.grouped(self._track_duration_field).items():
+            rotting_records = records.filtered(lambda record: fields.Datetime.now() > record.date_last_stage_update + timedelta(days=stage.rotting_threshold_days))
+            rotting_records.is_rotting = True
+            for record in rotting_records:
+                record.rotting_days = (fields.Datetime.now() - record.date_last_stage_update).days
+            clean_records += (records - rotting_records)
+        clean_records.is_rotting = False
+        clean_records.rotting_days = 0
+
+    def _search_is_rotting(self, operator, value):
+        """
+        :param operator
+        :param value
+        :return domain
+
+        Override this search method to complete the search domain for is_rotting field
+        """
+        if operator not in ['in', 'not in']:
+            raise UserError(_('Operation not supported'))
+        if not self._is_rotting_enabled():
+            return [fields.Domain.FALSE]
+
+        # review note:
+        # hey wouldn't it be cool if query objects actually gave you access to their alias table names?
+        # I am currently building on data whose alias names I don't know for sure and have to infer based on how
+        # they're build at time of writing, but if the alias names are then determined by something that isn't make_alias()
+        # in the future then this implementation breaks
+        # I think it would be neat to have this alias table for this reason...
+        # Are we allowed to write changes to framework? Or otherwise suggest them?
+        base_query = self._search(self._get_rotting_domain())
+        # Use the same alias as the base query to avoid non-matching alias issues
+        # Note that query objects do not make their alias list available
+        stage_id_alias_name = base_query.make_alias(self._table, self._track_duration_field)
+
+        # Items with a date_last_stage_update inferior to that number of months will not be returned by the search function.
+        max_rotting_months = self.env['ir.config_parameter'].sudo().get_param('crm.lead.rot.max_months', default=12)
+        max_rotting_months = int(max_rotting_months)  # safety/convenience, as this is a parameter without a matching settings entry
```

Comment:
One-line stuff when possible :) (also don't mix dots and underscores, that's bad, like ir.config_parameter, who invented that (I know who did it))
```suggestion
        max_rotting_months = int(self.env['ir.config_parameter'].sudo().get_param('crm.lead.rot.max.months', default=12))
```